src/sentry/api/endpoints/team_members.py

```diff
@@ -3,23 +3,54 @@
 from rest_framework.response import Response
 
 from sentry.api.bases.team import TeamEndpoint
-from sentry.api.serializers import serialize
-from sentry.models import InviteStatus, OrganizationMember
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import InviteStatus
+from sentry.models.organizationmemberteam import OrganizationMemberTeam
+
+
+@register(OrganizationMemberTeam)
+class DetailedOrganizationMemberTeamSerializer(Serializer):
+    def __init__(self, *args, **kwargs):
+        self.team = kwargs.pop("team", None)
+        super().__init__(*args, **kwargs)
+
+    def get_attrs(self, item_list, user, **kwargs):
+        attrs = {}
+        for org_member_team in item_list:
+            attrs[org_member_team] = {
+                "org_member": org_member_team.organizationmember,
+                "user": org_member_team.organizationmember.user,
+            }
+        return attrs
+
+    def serialize(self, obj, attrs, user):
+        org_member = serialize(attrs["org_member"])
+        org_member["user"] = serialize(attrs["user"])
+        org_member["teamRole"] = obj.role
+        org_member["teamSlug"] = self.team.slug
+        return org_member
 
 
 class TeamMembersEndpoint(TeamEndpoint):
     def get(self, request: Request, team) -> Response:
-        queryset = OrganizationMember.objects.filter(
-            Q(user__is_active=True) | Q(user__isnull=True),
-            organization=team.organization,
-            invite_status=InviteStatus.APPROVED.value,
-            teams=team,
-        ).select_related("user")
-
-        member_list = sorted(
-            queryset, key=lambda x: x.user.get_display_name() if x.user_id else x.email
+        queryset = (
+            OrganizationMemberTeam.objects.filter(
+                Q(organizationmember__user__is_active=True)
+                | Q(organizationmember__user__isnull=True),
+                organizationmember__organization=team.organization,
+                organizationmember__invite_status=InviteStatus.APPROVED.value,
+                team=team,
+            )
+            .prefetch_related("organizationmember")
+            .prefetch_related("organizationmember__user")
         )
 
-        context = serialize(member_list, request.user)
+        serializer = DetailedOrganizationMemberTeamSerializer(team=team)
 
-        return Response(context)
+        # x["user"] may be None as invited members will not have a linked user
+        members = serialize(list(queryset), request.user, serializer=serializer)
+        result = sorted(
+            members,
+            key=lambda x: (x["user"]["name"] or x["user"]["email"]) if x["user"] else x["email"],
+        )
+        return Response(result)
```

Comment:
Should we use pagination here? I guess it was already like this, but seems like this could be arbitrarily large.